experimental/relativetime/src/provider.rs

```diff
@@ -0,0 +1,99 @@
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_relativetime::provider)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct RelativeTimePatternsV1<'data>(
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    // Contains patterns in order second, minute, hour, day, week, month, quarter, year
+    pub [RelativeTimePattern<'data>; 8],
+);
+
+impl<'data> RelativeTimePatternsV1<'data> {
+    /// Create a new RelativeTimePatternsV1
+    pub fn new(patterns: [RelativeTimePattern<'data>; 8]) -> Self {
+        Self(patterns)
+    }
+}
+
+/// A single relative time pattern.
+#[derive(Debug, Clone, Default, PartialEq, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_relativetime::provider)
+)]
+#[yoke(prove_covariance_manually)]
+pub struct RelativeTimePattern<'data> {
+    /// The display name of the pattern
+    pub display_name: Option<Cow<'data, str>>,
+    /// Mapping for relative time fields.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub relatives: ZeroMap<'data, i8, str>,
+    /*
+    // Plural rules mapping for past.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_zero: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_one: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_two: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_few: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_many: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub past_other: Option<SingularSubPattern<'data>>,
+    // Plural rules mapping for future.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_zero: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_one: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_two: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_few: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_many: Option<SingularSubPattern<'data>>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub future_other: Option<SingularSubPattern<'data>>,
+    */
+}
+
+/// Singular substitution pattern string.
+#[derive(Debug, Clone, Default, PartialEq, yoke::Yokeable, zerofrom::ZeroFrom)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_relativetime::provider)
+)]
+pub struct SingularSubPattern<'data> {
```

Comment:
Suggestion: I'd probably put the 6 forms inside of this as individual fields and then have 2 instances above, one for future and one for past.

Reply:
Added a new struct `PluralRulesCategoryMapping` for this